domain/blockdag/blocknode.go

```diff
@@ -58,6 +77,9 @@ type blockNode struct {
 	// hundreds of thousands of these in memory, so a few extra bytes of
 	// padding adds up.
 
+	// dag is the blockDAG in which this node resides
+	dag *BlockDAG
```

Comment:
The spaghetti thickens...

Reply:
Is much better then passing node.blabla(dag) all the time.